server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -61,8 +61,6 @@
    * This method may trigger a {@link #persistAll(Committer)} using the supplied Committer. If it does this, the
    * Committer is guaranteed to be *created* synchronously with the call to add, but will actually be used
    * asynchronously.
-   * <p>
-   * The add, clear, persist, persistAll, and push methods should all be called from the same thread.
```

Comment:
Is something making sure that "add" won't be called while these other methods are being called? What I'm concerned about is preventing a case where the Committer passed to "publish" is out of sync with what's being "add"ed.

I think if we need to relax the constraint that the methods are all called from the same thread, we should at least require they not be called from different threads at the same time. Maybe we can enforce this with synchronization? Otherwise, I'm concerned that commit metadata will get out of sync with the data itself.

Do you have any thoughts on this?